core/src/com/unciv/models/ruleset/unique/UniqueType.kt

```diff
@@ -756,6 +756,7 @@ enum class UniqueType(
     /** Can be placed upon both units and as global */
     TriggerUponConqueringCity("upon conquering a city", UniqueTarget.TriggerCondition, UniqueTarget.UnitTriggerCondition),
     TriggerUponFoundingCity("upon founding a city", UniqueTarget.TriggerCondition),
+    TriggerUponBuildingImprovement("upon building [improvementFilter]", UniqueTarget.TriggerCondition, UniqueTarget.UnitTriggerCondition),
```

Comment:
Definitely change this to "upon building a [improvementFilter] improvement", looks too much like "upon constructing" below and could easily confuse both players and modders